test/fortran/f_c_string/Test_f_c_string.c

```diff
@@ -0,0 +1,16 @@
+#include "Test_f_c_string.h"
+#include <ctype.h>
+#include <stdio.h>
+
+void to_upper(char **c_string) {
```

Comment:
From Fortran, if you pass a `type(c_ptr)` by `value`, then you can avoid an extra `*` here. Similarly for the `to_upper_array` function.

Reply:
If I use `value`, I would need to change the intent of `c_string` and `c_string_array` from `intent(inout)` to `intent(in)`. Since `to_upper` and `to_upper_array` modify their arguments, would it still make sense to mark these arguments as `intent(in)`?

Reply:
While `to_upper` modifies the memory pointed to by `c_string`, it doesn't modify the pointer itself. So I think `intent(in)` should be work.